source/ptreferential/ptreferential_ng.cpp

```diff
@@ -294,22 +294,23 @@ struct Eval : boost::static_visitor<Indexes> {
         } else if (f.type == "disruption"
                    && ((f.method == "tag" && f.args.size() == 1) || (f.method == "tags" && f.args.size() >= 1))) {
             indexes = get_impacts_by_tags(f.args, data);
-        } else if ((f.method == "since" && f.args.size() == 2) || (f.method == "until" && f.args.size() == 2)
-                   || (f.method == "between" && f.args.size() == 3)) {
+        } else if ((f.method == "since" && f.args.size() == 1 + (f.type == "vehicle_journey"))
+                   || (f.method == "until" && f.args.size() == 1 + (f.type == "vehicle_journey"))
```

Comment:
you use a`+` in a logic condition ? isn't it `&&` ?

Reply:
Actually it's on purpose, I implicitly (:see_no_evil:) cast the bool to an int to add one more mandatory arg.
Maybe I should cast explicitly and add a comment ?

Reply:
I don't know how to change the condition in a `real logic condition style`, but it is not obvious to read.
You should probably find a way.

Reply:
TBH, I don't quite understand what's happening here by reading the code :(
What exactly are you trying to express here ?

Reply:
Ok, I think I got it. It means that you expect one more arguments in the case of a `vehicle_journey` isn't it ?